apps/projects/app/App.js

```diff
@@ -57,247 +59,176 @@ const getTabs = ({ repoCount }) => {
-  }
+  })
 
-  handlePopupMessage = async message => {
+  const handlePopupMessage = async message => {
     if (message.data.from !== 'popup') return
     if (message.data.name === 'code') {
       // TODO: Optimize the listeners lifecycle, ie: remove on unmount
-      window.removeEventListener('message', this.messageHandler)
+      window.removeEventListener('message', handlePopupMessage)
 
       const code = message.data.value
       try {
         const token = await getToken(code)
-        this.setState(
-          {
-            githubLoading: false,
-          },
-          () => {
-            this.props.api.cache('github', {
-              event: REQUESTED_GITHUB_TOKEN_SUCCESS,
-              status: STATUS.AUTHENTICATED,
-              token,
-            })
-          }
-        )
+        setGithubLoading(false)
+        api.cache('github', {
+          event: REQUESTED_GITHUB_TOKEN_SUCCESS,
+          status: STATUS.AUTHENTICATED,
+          token,
+        })
       } catch (err) {
-        this.setState(
-          {
-            githubLoading: false,
-          },
-          () => {
-            this.props.api.cache('github', {
-              event: REQUESTED_GITHUB_TOKEN_FAILURE,
-              status: STATUS.FAILED,
-              token: null,
-            })
-          }
-        )
+        setGithubLoading(false)
+        api.cache('github', {
+          event: REQUESTED_GITHUB_TOKEN_FAILURE,
+          status: STATUS.FAILED,
+          token: null,
+        })
       }
     }
   }
 
-  handleMenuPanelOpen = () => {
-    window.parent.postMessage(
-      { from: 'app', name: 'menuPanel', value: true },
-      '*'
-    )
-  }
-
-  changeActiveIndex = activeIndex => {
-    this.setState({ activeIndex })
+  const changeActiveIndex = activeIndex => {
+    setActiveIndex({ activeIndex })
   }
 
-  closePanel = () => {
-    this.setState({ panel: null, panelProps: null })
+  const closePanel = () => {
+    setPanel(null)
+    setPanelProps(null)
   }
 
-  setPanel = {
-    setActivePanel: p => this.setState({ panel: p }),
-    setPanelProps: p => this.setState({ panelProps: p }),
+  const configurePanel = {
+    setActivePanel: p => setPanel(p),
+    setPanelProps: p => setPanelProps(p),
   }
 
-  handleGithubSignIn = () => {
+  const handleGithubSignIn = () => {
     // The popup is launched, its ref is checked and saved in the state in one step
-    this.setState(({ oldPopup }) => ({
-      popup: githubPopup(oldPopup),
-      githubLoading: true,
-    }))
+    setGithubLoading(true)
+
+    setPopupRef(githubPopup(popupRef))
+
     // Listen for the github redirection with the auth-code encoded as url param
-    window.addEventListener('message', this.handlePopupMessage)
+    window.addEventListener('message', handlePopupMessage)
   }
 
-  handleSelect = index => {
-    this.changeActiveIndex({ tabIndex: index, tabData: {} })
+  const handleSelect = index => {
+    changeActiveIndex({ tabIndex: index, tabData: {} })
```

Comment:
As per comment above. Actually... changing this could be a part of bigger refactoring: moving Apollo and GQL calls to workers, along with filtering. Then this code could be much improved.